--- a/prepare_mkl.sh
+++ b/prepare_mkl.sh
@@ -75,14 +75,14 @@ MXNET_ROOT=`dirname $0`
 USE_MKLML=0
 # NOTE: if you update the following line, please also update the dockerfile at
 # tests/ci_build/Dockerfile.mkl
-VERSION_MATCH=20170908
+VERSION_MATCH=20171227
 PLATFORM=$(uname)
 if [ $PLATFORM == "Darwin" ]; then
     INFIX=mac
 elif [ $PLATFORM == "Linux" ]; then
     INFIX=lnx
 fi
-ARCHIVE_BASENAME=mklml_${INFIX}_2018.0.20170908.tgz
+ARCHIVE_BASENAME=mklml_${INFIX}_2018.0.1.20171227.tgz
 MKL_CONTENT_DIR=`echo $ARCHIVE_BASENAME | rev | cut -d "." -f 2- | rev`
-MKLURL="https://github.com/01org/mkl-dnn/releases/download/v0.10/$ARCHIVE_BASENAME"
+MKLURL="https://github.com/intel/mkl-dnn/releases/download/v0.12/$ARCHIVE_BASENAME"
 # there are diffrent MKL lib to be used for GCC and for ICC
